service-base/src/main/java/org/eclipse/hono/service/AbstractAdapterConfig.java

```diff
@@ -163,4 +166,64 @@ public ConnectionFactory registrationServiceConnectionFactory() {
     public HonoClient registrationServiceClient() {
         return new HonoClientImpl(vertx(), registrationServiceConnectionFactory());
     }
+
+    /**
+     * Exposes configuration properties for accessing the credentials service as a Spring bean.
+     * <p>
+     * Sets the <em>amqpHostname</em> to {@code hono-device-registry} if not set explicitly (reflecting that Hono is
+     * providing the credentials API as part of the device registry component).
+     *
+     * @return The properties.
+     */
+    @Qualifier(CredentialsConstants.CREDENTIALS_ENDPOINT)
+    @ConfigurationProperties(prefix = "hono.credentials")
```

Comment:
this will not work because the bean will be created in any case,  regardless of whether environment variables starting `HONO_CREDENTIALS_` exist or not. In the latter case the properties object will simply have default values for its fields. You need to create the bean on condition of the existence of such an environment variable only. Check the code we had originally when the device registry was part of the Hono Server for the blueprint for this.